### DISCUSSION TITLE
Version 2.1 Beta 1

### DISCUSSION THREAD

**Discussion by vkbo** [vkbo/novelWriter#1491] [General]
### Release Notes

This is a beta release of the next release version, and is intended for testing purposes. Please be careful when using this version on live writing projects, and make sure you take frequent backups.

Please check the changelog for an overview of changes. The full release notes will be added to the final release.

### Detailed Changelog

**Usability**

* When the main window is resized, the change in size is only assigned to the editor and viewer. To resize the project tree area, its slider needs to be moved. PR #1388.
* The default text font on MacOS is now Helvetica instead of Courier. Issue #1463. PR #1479.
* Backup files now contain the project name again. Issue #1476. PR #1484.
* The backup success dialog now displays the file size of the backup file. Issue #1453. PR #1484.
* New root folders in the Project Tree now appear next to the root folder of the item selected when the request to make the root folder was made. Previously, it would appear at the bottom of the Project Tree. Issue #1259. PR #1487.

**Features**

* A new Manuscript Build Tool has been added. The new tool allows for detailed handling of which documents are included in a build, as well as a much better tool for formatting headers. It also allows for saving multiple build presets. PRs #1389 and #1466. Issues #971, #1315 and #1448.
* Exported ODT documents now have an accessible style for scene separators. It is also possible to define page size and margin sizes from the new build tool. Issue #622. PR #1477.
* A proper light colour theme has been added. The default theme will usually default to light colours, but in Qt6 it will not depending on host OS settings, so creating a proper light theme is needed. This also allows for some tweaking of the colours. The contrast of the dark theme has been improved a bit as well, and a default icon theme is now selected based on the lightness of the background if an icon theme is not specified in the theme definition file. Issues #1472 and #1473. PR #1475.
* Documents, folders and root folders can now be duplicated from the Project Tree, including all child elements. The duplicated content is stored next to the source items, and can then be moved to wherever the user wants a copy. Issue #1469. PR #1480.
* A set of new keyboard shortcuts have been added to make some types of navigation in the Project Tree easier. `Alt+Up` and `Alt+Down` will move between sibling items in the tree, skipping child items. `Alt+Left` will move to the parent of the selected item without triggering the collapse of the node like the `Left` key does. `Alt+Right` does the reverse, and both expands the node and moves to the first child in one click. Issue #1348. PRs #1488 and #1489.

**Packaging and Installation**

* Support for Python 3.7 is no longer maintained, but has not officially been dropped. It is expected to be dropped for the final release of 2.1. PR #1452.
* The `lxml` package has been removed from the source code, dropping it as a dependency of novelWriter. The standard Python `xml` library is used instead. The standard library is somewhat limited, which is why it wasn't originally used, but when dropping support for Python 3.7, it is now good alternative. Issue #1257. PR #1452.
* The `setup.py` file has been removed. The custom packaging utilities in the old `setup.py` file are now available in `pkgutils.py` instead. Issues #1437 and #1438. PR #1483.

**Code Improvements**

* All imports of modules are now direct imports instead of going via init files. All subfolder init files have been reduced to empty files. PR #1262.
* The mocking of the main config object in the test suite has been rewritten to be easier to deal with when writing tests. The new approach also removes the need to access the config object via an attribute in many classes, and is now instead accessed directly. This should give a tiny performance boost as a bonus. PR #1447.
* The building of manuscript documents from novelWriter source text is now handled by a core builder class, thus separating it from any GUI module. Previously, this was baked into the build tool. PR #1316.
* SVG icons have been optimised in terms of storage size and object complexity. PR #1456.
* The file names for the project meta data files have been simplified and references to legacy formats removed. The wordlist has been converted to a JSON file, and the session log to a JSON Lines file. All old files are renamed or converted on the fly when opening the project. PR #1464.
* The core project item and tree classes have been modified to improve how items, and in particular, orphaned items are handled. These are mostly internal changes that simplifies how items are accessed in the source code. Issue #1481. PR #1482.
* Many of the above PRs add type annotations to classes and functions in the source code. These will be added gradually to the entire source code going forward.

<hr /><em>This discussion was created from the release <a href='https://github.com/vkbo/novelWriter/releases/tag/v2.1b1'>Version 2.1 Beta 1</a>.</em>

**Comment by novakwh** [comment]
On Mac: in the Build Tool, either editing the default My Manuscript build or adding a new build, under Format, Font, selecting a new Font and pressing OK in the font dialog causes the Build window to simply shut down (without saving any of the changes made).

**Reply by vkbo** [reply]
Thanks for letting me know. I will investigate.

**Reply by vkbo** [reply]
Could you check if this is also the case for selecting a font in Preferences? This seems to be isolated to Mac, as I could not reproduce this on Linux.

**Reply by novakwh** [reply]
Font selection in preferences (both in general and documents) seems to work fine.

**Reply by vkbo** [reply]
I managed to reproduce it on a Mac VM. Curious. I will investigate.

By the way, did you have any issues installing the new DMG file? I had a lot of trouble. I'm not a Mac-user so I may have made a mistake.

**Reply by vkbo** [reply]
The issue has been fixed, as well as some version number issues in the DMG package plist file. The new DMG is versioned 2.1b1-post1. Please download and install the new file,

**Comment by HeyMyian** [comment]
:smiley: New stuff!!

Should I create a reply for each thing I notice or would you rather I gather everything in one message?

There seems to be no fallback for new icons. I'm using a custom dark theme with some custom icons and the new icons are left blank in the UI.
![Screenshot_2023-07-30_15-41-07](https://github.com/vkbo/novelWriter/assets/60368695/97566698-5ff1-4533-936c-c9a1276f4bd1)

**Reply by vkbo** [reply]
If you have a custom icon theme, you need to add the missing icons unfortunately. There is only fallback for whole themes, not individual icons.

**Reply by vkbo** [reply]
If you find any issues that are bugs, or have some usability suggestions, feel free to open feature tickets or bug report. You can collect suggestions for specific tools or features.

It's also fine to post them here for discussion, but I can't use the discussion for tracking stuff that needs fixing.

**Comment by HeyMyian** [comment]
The build preview used to be a file written to the "cache" folder. I never used the preview feature because it was kind of a pain to exclude that file from the backups. How/where is the preview stored in 2.1?

**Reply by vkbo** [reply]
I moved it to the app data folder, which on Linux is `~/.local/share/novelwriter/cache`. Each project now has a uuid associated with it, and the cache file is named `build_{uuid}.json`.

The project `cache` folder has been removed. I never liked bloating the project folder with these files either.

**Reply by HeyMyian** [reply]
Nice :D I was going to suggest allowing the deletion of the preview file, but this is even better.

**Comment by novakwh** [comment]
"By the way, did you have any issues installing the new DMG file? I had a lot of trouble. I'm not a Mac-user so I may have made a mistake."

I received a number of security warnings and had to try to open it several times before the system would allow me to open it, but I guess I'm used to that with Mac.

**Reply by vkbo** [reply]
There was an issue with the version number yes, so I have uploaded a new DMG file. This one installed smoothly.

**Comment by HeyMyian** [comment]
I don't find the new Manuscript Build tool very intuitive -- usable and functional, yes. Now YMMV and maybe I'll come to love it soon. But these are my first impressions, maybe they help, maybe I'm obtuse, maybe you can also just disregard them if you have a different opinion.

### Filters

My main confusion is with the "Filters" setting, because I don't understand what the orange filter icon *means* if it's there or not. There seem to be two overlapping functionalities: The filter icon, and on the right hand side the switches for "Document Types". Only if the respective document type is activated, are the documents in the tree able to be filtered. Filtered for what? "Included" and "Excluded" as well as the "Root Folder" switches are easy enough to understand. Although, that's yet *another* type of filter, this time also visually removing the unselected folders from the tree.

It doesn't help, visually speaking, that folders in the tree can't get an icon, giving them the idea that in its place is a Document of a Document Type that isn't selected with the switches. I would suggest formatting folders in the tree differently, although seeing as the UI right now is consistent with the other trees, maybe the folders could get an icon of their own? Maybe a different background color? Maybe folders should be able to get icons anyway, valid for their content? Idk. It's just a source of my feeling of "what is going on here".

Left hand side tabs: The "Options" heading looks a bit like an inactive tab, not a headline. I might or might have not tried a couple of times to click on it :no_mouth:. "Preferences" uses different tabs, any reason they aren't reused here?

### Headings

With "Headings" I understand these are multi-line textareas now instead of the single-line input fields to edit the headings. I'm not trained in UI design or UX, but intuitively, I would simply swap the input fields out for textareas, and the "Edit" button for the two buttons "Insert" and "Apply" for each textarea. So instead of having stacked five input fields + one textarea, it would be five textareas. There should be enough vertical space for it?

I would also suggest to add, maybe on hover over the headline type or in brackets, the markdown symbols for each heading (##, !##, ...)

Line breaks here are carried over as `//` into the input field and into the final file.

### Format

* "Font Size" and "Line Height" have the units pt and em respectively. Those have different visual widths, which results in the input boxes not being aligned. Maybe the units in monospace font? (Thinking of it, people might not know what kind of unit "em" is. Using "%" instead might be more intuitive.)
* "Text Options" has different line heights than the rest of the settings. I understand this is because the switches don't need as much space vertically, but it creates a visual imbalance.
* I also just see that the units in the right column are controlled by the dropdown, and not written behind the input boxes. Maybe add them anyway and they are changed on the fly based on what is selected in the dropdown.
* Selecting a "Page Size" locks in the values. Instead, the dropdown should change to "Custom" when the values are changed.

### Output

In "Output", the headline "Open Document" is super ambiguous by itself. Maybe add .otf or "format". 

### General

Something that confused me in the old Build tool too: Some of the options in "Format" and "Output" have no effect on some of the "Output Formats" (I mean file types). Which... of course they don't, because there is no Font Size in json and no Page Size in html. I find this added to my general feeling of being flummoxed with the old tool, not knowing what certain settings did, not changing anything after my initial setup, using it only with velvet gloves on. I do think the new tool is objectively better, but I also think it could be improved still? Besides my notes with the suggestions above, I wouldn't know exactly how to get rid of that fuzzy feeling of "all of this isn't as clear as I would like, I don't know". Maybe it _is_ me.

---

Thanks for the whole thing btw, I realize there is only gripes here, not praise, but genuinely, thanks. novelWriter is the only project where I participate on GitHub because I like it so much. I like LibreOffice and Nextcloud too, but I really couldn't care less about what their folks are doing in their repos.

**Reply by vkbo** [reply]
Yes, I was wondering how intuitive the filter feature was. That's why I'm doing this beta release, to iron out usability. Due to the many options requested, it was bound to be more complex. This is a case of not being able to have both simplicity and lots of features. But I'm happy to make improvements based on first impressions (which I obviously lack, having made this and all).

### Filters

> My main confusion is with the "Filters" setting, because I don't understand what the orange filter icon _means_ if it's there or not. There seem to be two overlapping functionalities: The filter icon, and on the right hand side the switches for "Document Types". Only if the respective document type is activated, are the documents in the tree able to be filtered. Filtered for what? "Included" and "Excluded" as well as the "Root Folder" switches are easy enough to understand. Although, that's yet _another_ type of filter, this time also visually removing the unselected folders from the tree.

The filter icon indicates that the document is included or excluded by the options on the right. The specific "included" and "excluded" options overrides whatever the filter is set to.

The reason root folders are removed when you switch them off is that it would be confusing if you could then set documents in them to be explicitly included or excluded, or would try to. Maybe fading them out would be a better option, but in a large project, I expected it would be better to not have them cluttering up the UI.

**Suggestions:**

* Would it be better if there was another row of icons that indicated if a document was included or not, so that the filtered/included/excluded icons were always present? This was my original design, but I though it was a little messy, Especially since there also is the active/inactive icon. I wasn't sure how to design the third icon.
* Maybe the filtered button below the tree should instead be a reset button to cancel the included or excluded setting?
* The "Document Types" and "Root Folders" text could definitely be improved. How about "Document Filters" and "Root Folder Selection" or something?

An alternative design for the filter tab that I considered, is the following (roughly):

Each document is included or excluded directly with a checkbox on the left, and the "Document Types" options are instead buttons that toggle or untoggle all items of that specific type. So they would be "Select/unselect all novel documents", "Select/unselect all project notes" and the inactive one would be the same as it is, as the other two decides whether it is selected or not.

The root folders switches will stay as they are, but relabelled. Although toggling them makes it fairly obvious what they do.

This would make the filter tab a lot easier to understand, but also a lot less flexible. I'm not sure how much control users actually want. Personally, I need the option to export specific root folders or chapters, and pretty much nothing else.

I do plan to add a quick export feature directly in the project tree that can be used on single documents or documents/folders with sub-items. Maybe that would better serve the last use case, and the filter tab can just be simplified.

**That said ...**

The main reason the filter approach was used is that it will automatically handle new documents added. If they had to manually be selected, then what would you want to do with new documents? Always select them? Always select novel documents, but not notes? With this setup, it's clear.

Also, the build system is designed to later be exposed to the command line, where you would only have the filter settings.

> It doesn't help, visually speaking, that folders in the tree can't get an icon, giving them the idea that in its place is a Document of a Document Type that isn't selected with the switches. I would suggest formatting folders in the tree differently, although seeing as the UI right now is consistent with the other trees, maybe the folders could get an icon of their own? Maybe a different background color? Maybe folders should be able to get icons anyway, valid for their content? Idk. It's just a source of my feeling of "what is going on here".

Folders cannot be included at all, which is why they cannot have the icon. The icon indicates whether an item is included in the build or not.

> Left hand side tabs: The "Options" heading looks a bit like an inactive tab, not a headline. I might or might have not tried a couple of times to click on it no_mouth. "Preferences" uses different tabs, any reason they aren't reused here?

This is a fairly standard design, used in other applications. I basically just stole it from another Electron-based app. I plan to use it for the other tabbed dialogs later, but they need to be redesigned first.

### Headings

> With "Headings" I understand these are multi-line textareas now instead of the single-line input fields to edit the headings. I'm not trained in UI design or UX, but intuitively, I would simply swap the input fields out for textareas, and the "Edit" button for the two buttons "Insert" and "Apply" for each textarea. So instead of having stacked five input fields + one textarea, it would be five textareas. There should be enough vertical space for it?

The reason for this is purely technical. The text area can have syntax highlighting (it is essentially identical to the main document editor). The input fields cannot. I'm not sure what the resource overhead would be to duplicate the text area 5 times over, but I can try it out.

The reason I went for the layout is also aesthetical, as everything lines up in a way that it's easy to read. That comes at a cost here in terms of editing, sure.

**Suggestions:**

* I can change the text boxes to be two-line text areas. I would then remove the edit button, and just let them be editable by default. The "Insert" button would then need to be applied to the currently active edit box, but that shouldn't be a problem.

> I would also suggest to add, maybe on hover over the headline type or in brackets, the markdown symbols for each heading (##, !##, ...)

Sure, good idea.

> Line breaks here are carried over as `//` into the input field and into the final file.

Ok, that's a bug. I will fix that.

### Format

> * "Font Size" and "Line Height" have the units pt and em respectively. Those have different visual widths, which results in the input boxes not being aligned. Maybe the units in monospace font? (Thinking of it, people might not know what kind of unit "em" is. Using "%" instead might be more intuitive.)

This is the same layout system used in Preferences. I'm considering redesigning the underlying layout, but not until I redesign Preferences.

> * "Text Options" has different line heights than the rest of the settings. I understand this is because the switches don't need as much space vertically, but it creates a visual imbalance.

Yes, this is standard Qt behaviour. I haven't decided what and if to do anything about this. This too is part of the Preferences updates, since the same layout is used here.

> * I also just see that the units in the right column are controlled by the dropdown, and not written behind the input boxes. Maybe add them anyway and they are changed on the fly based on what is selected in the dropdown.

Sure, that makes sense.

> * Selecting a "Page Size" locks in the values. Instead, the dropdown should change to "Custom" when the values are changed.

Yeah, that should be fairly easy to change.

### Output

> In "Output", the headline "Open Document" is super ambiguous by itself. Maybe add .otf or "format".

Good idea.

### General

> Something that confused me in the old Build tool too: Some of the options in "Format" and "Output" have no effect on some of the "Output Formats" (I mean file types). Which... of course they don't, because there is no Font Size in json and no Page Size in html. I find this added to my general feeling of being flummoxed with the old tool, not knowing what certain settings did, not changing anything after my initial setup, using it only with velvet gloves on. I do think the new tool is objectively better, but I also think it could be improved still? Besides my notes with the suggestions above, I wouldn't know exactly how to get rid of that fuzzy feeling of "all of this isn't as clear as I would like, I don't know". Maybe it _is_ me.

The old tool was crammed for space, and with an increasing number of feature requests, there was no other option than to split the build settings out. I did consider making the preview area tabbed so that the settings could be embedded there instead of in a new window, I'm still wondering if that would be easier. The downside is that the layout gets very messy when there are even more columns of controls. Also, now you can have multiple build settings dialogs open to compare or copy over settings. (There is a bug now that allows you to open the same build twice, but I'm fixing that for next release.)

> Thanks for the whole thing btw, I realize there is only gripes here, not praise, but genuinely, thanks. novelWriter is the only project where I participate on GitHub because I like it so much. I like LibreOffice and Nextcloud too, but I really couldn't care less about what their folks are doing in their repos.

😃

**Reply by vkbo** [reply]
I'm making some changes to the Filter tab, but I am a little reluctant to fundamentally change how it works. Mostly because it will reduce its functionality and features. I think the better approach is to make it easier to use through redesigning the interface itself.

I'm considering adding a per-component section to the documentation where I can instead explain how to use the GUI tools. Perhaps with a direct link button from the dialog itself. Do you think that would help?

Also, perhaps finding a different icon for the filter may be useful. The current icon is a standard filter icon, but perhaps people don't recognise it as such?

**Reply by novakwh** [reply]
I will pipe in here for just a second since Veronica had mentioned she wasn't sure how much control users want, so all I can do is offer my perspective as a user. 

I did find the new builder a little confusing, but was able to figure it out fairly quickly. Could the interface be a little more smooth? Yes, but honestly it's not a dealbreaker for me. Because strictly in terms of functionality, I love it -- it is MUCH easier to use, in my opinion, than the current builder.

My particular use case: I simply need to export novel chapters and scenes, sometimes an entire manuscript, but not usually; instead it is almost always just a small number of chapters and/or scenes. So what I want to do is very quickly select which chapters and/or scenes to export to ODT (sometimes pdf, but usually ODT), make sure that the headings and scene separators are correct , and press build. The current system, of changing the active and inactive flags on the project screen, is soul-deadening; I just wind up exporting the whole manuscript each time and then searching through it for what I need to review. 

The one major suggestion I have for the "Filters" might be to switch that window so that the folders and document types are on the left so that they are the primary focus of the screen. Then you could toggle each folder on or off, just as the beta builder allows now, but when you highlight a folder that is turned on, the right side of the screen would list all of the documents inside that folder (that match the toggled document types) so that each can be quickly selected for inclusion, or deselected for exclusion. (Again, toggling the folder itself off would deselect all of them). 

Basically, I just think it would be slightly less confusing if the view of the builder dialog was switched from right to left, and that instead of documents appearing/disappearing all in one window based on the toggle, instead you can toggle the folders off/on while also selecting the folders that are turned on in order to see the documents inside for greater granularity of selection. 

Even if you keep everything the way it is now in the beta, I think it would benefit from having the folder and document types be the "focus" (on the lefthand side) so that you immediately know that making those selections is key (right now my eyes are drawn to the documents list on the left and it took me a while to realize that the selections of types and folders were so important, and that I didn't have to scroll all the way through every single document in all the folders, since I could turn off certain types and folders). 

Finally, "filtered" is really confusing to me as a term; it took me a long time to figure out what that meant. I think it's probably a useful thing to have (the override) but not so useful that it should be first in the list? Also perhaps a better word to describe what it is? Maybe change included to "active" and filtered to "always include" or something like that, though I'm not sure, and differentiate it from the filter icon. I guess I am still a little confused about the filtered, since it shows up automatically based on selections, but you can also override the selection to include anyway -- again, it just seems confusing to me. 

None of these are dealbreakers, like I said, I prefer this builder dramatically to the existing one, but it could be made a little clearer. Sorry if my suggestions are too confusing, but I'm a writer, not a coder or UX designer. :-)

**Reply by vkbo** [reply]
Thanks for the feedback!

So, what I *do* want to add, and that I briefly mentioned above, is a way to export a group of items directly from the project tree without having to open the build tool. This is why the writing of the final document itself is in a separate dialog, and shows you a list of the documents included.

The idea is to select a build definition in the build tool as a default build (which will be the "My Manuscript" options if none are explicitly selected), and you will be able to right-click any document or folder in the project tree and run that default build on those files. If you do this on a folder or document with sub items, they will all be exported as long as they are marked as "active".

Anything more complex than that will require opening the full build tool.

As for your suggestions: Switching the order may help. I'll test it. The reason it is the way it is, is so it mimics the main window, where the tree of items is on the left. But I can give it a test.

About a better term for "filter", I'm all for that. I just couldn't think of anything. Feel free to suggest something else. Essentially, any way to indicate which items are included based on the switches, or explicitly included or excluded as an override, that is clearer than now would be useful. I'd just like to retain the current features, regardless of what we call them.

**Reply by vkbo** [reply]
Ok, I've made some improvements to the filter tab based on @HeyMyian's suggestions:

![image](https://github.com/vkbo/novelWriter/assets/1619840/4267544d-addf-4748-a165-a46dd56d79da)

I also tested switching the tree and options to opposite sides, but I don't think that worked very well. The dialog seems unbalanced as the options list is a little short and the tree is big. I think it's a better approach to just make it clearer what everything does.

Also, there aren't that many controls on that tab really, so I think the challenge is to make it easier to understand.

**Reply by vkbo** [reply]
Here's another suggestions.

* I've renamed the tab from "Filters" to "Selection".
* I've replaced the filter icon with a new symbol that matches the arrow used for builds.
* The filtered button now has a reset/undo icon instead. The behaviour is unchanged, but it is now more obvious that it clears the "included"/"excluded" options. The old filter icon sort of implied that the filter icon would appear on the item, which it doesn't necessarily do.

The three buttons on the bottom have the following tooltips:

1. "Reset to default"
2. "Always included"
3. "Always excluded"

Each document in the tree will have one of the following tooltips:

1. "Included in manuscript"
2. "Excluded from manuscript"

![image](https://github.com/vkbo/novelWriter/assets/1619840/2e588f85-782d-444d-9ad5-8672f4629f0c)

How does this look?

**Reply by novakwh** [reply]
This makes sense to me, I think, with one suggestion for you to consider: if you moved the "Reset to Default" action to third in the list after "Mark selected as", it would make a little more sense to me. So the options at the bottom there would read as Mark selected as: Always included, or Always excluded, or Reset to default. I feel like that reset action is the least likely to be used when building (at least by me), but putting it first makes me stop and think about it for longer than I should.

**Reply by vkbo** [reply]
That's a good suggestion.

Actually, the original design had both icons and text on the buttons. Maybe in this particular instance, that would be useful to add back.

**Reply by HeyMyian** [reply]
I like the green arrow icon. The orange of the previously used filter icon might have had unwanted connotations (usually "warning"). Did you also change the padding of that column? For me there was more space between the active/inactive and the filter icon. I like the spacing in your screenshot more.

Perhaps a tooltip or a small text (small text like in the Preferences menu) for "Document Filters" and "Root Folders" on the right side: "default include/exclude". As I understand it, that's what these switches are primarily meant to do: automatically control the inclusion or exclusion of newly created documents.

While the "Manuscript Build Settings" window is open, the main novelWriter window is still editable - that means the project tree in the main window can be changed, and the changes aren't reflected in the build window. Duh. I think that's reasonable, but what might be useful is a context menu on the items of the build tree that allows to toggle Active/Inactive.

* As I discovered this, I was wondering if the build tool could be integrated into the main window as a sibling to the "Outline" view, i.e. not as a popup. It makes sense to me logically, because the project tree is already in that place. The "Novel Selection" dropdown could be used as build presets. But I'm brainstorming, there might be a reason why it's a popup.

The input label for "Name" on your screenshot I would suggest to change to "Build Name" for clarity.
It's a little visually noisy in that last line, things don't seem to align to the other elements above.

I like the arrows on the left menu (I think they're new now?).

What is your time scale for the full release of 2.1?

**Reply by vkbo** [reply]
> Did you also change the padding of that column? For me there was more space between the active/inactive and the filter icon. I like the spacing in your screenshot more.

No, it uses the same logic as the main project tree. It is the icon size + 6 pixels.

> Perhaps a tooltip or a small text (small text like in the Preferences menu) for "Document Filters" and "Root Folders" on the right side: "default include/exclude". As I understand it, that's what these switches are primarily meant to do: automatically control the inclusion or exclusion of newly created documents.

I'm not sure that helps all that much. I'm trying to not add too much text as well as everything has to be translated, and the translation tool is capped at a certain word count. I think playing with the settings will make it fairly easy to see what they do. They are also the exact same as on the old build tool.

> While the "Manuscript Build Settings" window is open, the main novelWriter window is still editable - that means the project tree in the main window can be changed, and the changes aren't reflected in the build window. Duh. I think that's reasonable, but what might be useful is a context menu on the items of the build tree that allows to toggle Active/Inactive.

A refresh button may be useful. The old build tool was also allowing editing the project with the tool open, and the content was refreshed when you pressed preview or build. Here it would make more sense to have a button, or maybe hook into a signal from the main project window.

> As I discovered this, I was wondering if the build tool could be integrated into the main window as a sibling to the "Outline" view, i.e. not as a popup. It makes sense to me logically, because the project tree is already in that place. The "Novel Selection" dropdown could be used as build presets. But I'm brainstorming, there might be a reason why it's a popup.

The main reason it isn't is that it isn't a "view", as in, not an alternate view of the project. I have "views" and "tools" in novelWriter. This is a tool. They are called on when needed. I plan to add more views as well.

> The input label for "Name" on your screenshot I would suggest to change to "Build Name" for clarity. It's a little visually noisy in that last line, things don't seem to align to the other elements above.

I don't really know where's a good place to put it. Perhaps it would be good to add a line between the settings and the bottom section.

> I like the arrows on the left menu (I think they're new now?).

Yes, I just added them as a visual indicator to separate them from the "Options" label.

> What is your time scale for the full release of 2.1?

I think I need to make a Beta 2 release with these changes. Maybe this coming week if I finish the changes. Then I will need a release candidate before full release. That's when I upload the translations and wait for the translation work. I usually give that 3-4 weeks.

I have a few features to add for the release candidate, but I may just push those to release 2.2, depending on how much work they are.

**Reply by vkbo** [reply]
I tested converting each heading into an editable area, but I don't think it looks very good. The old format may require two more clicks, but it's not the part you would need to change a lot (as opposed to the Selection tab).

![image](https://github.com/vkbo/novelWriter/assets/1619840/880ce650-baca-43ff-bab4-1f950584bac9)

**Reply by vkbo** [reply]
I'm adding a feature where if you select a single item in the list of project items on the Selection tab, then press one of the buttons below the tree, and that item is a folder or document with child items, the effect of the button is applied to all child items. For a multi-select, only the selected items are affected.

I'm not sure if people will react positively or negatively to this behaviour. It should make it easier to include/exclude chapters or whole folders.

**Reply by vkbo** [reply]
I've made a number of improvements in PR #1499. They are listed there. I'll upload a snapshot to Launchpad at least, but the rest will have to wait until I make the Beta 2 release later the coming week.

**Comment by xahodo** [comment]
YAY! Another release (even when it's a beta).

Keep up the good work! Really nice improvements!

**Reply by vkbo** [reply]
I'm almost ready with a Beta 2 now. I ended up doing a lot of refactoring of the code that handles the project data. Hopefully I'll do a release again this week.
